### DISCUSSION TITLE
Purge Database Data

### DISCUSSION THREAD

**Discussion by silicahd** [tailwindlabs/tailwindcss#8565] [Help]
I hava a project built in laravel. I use a drag and drop builder to build the front page modules. The code is saved in the database as HTML. This way on the front page I can do a foreach modules as module and output the code. The issue I have is that i cant minify the code in the database so I am stuck to using the tailwind CDN in production. Any ideas how I could minify the code in the database using npm run watch? 

I can minify the code in the files like this. 

`module.exports = {
    content: [
        './Modules/**/Resources/views/**/*.blade.php',
        './Modules/**/Resources/views/*.blade.php',
        './resources/views/**/*.blade.php',
        './resources/css/**/*.css',
        './src/**/*.{html,js}', './node_modules/tw-elements/dist/js/**/*.js'
  ],`

**Comment by pySilver** [comment]
Have you found a way to compile css out of dynamic HTML?

**Reply by silicahd** [reply]
I did not :( I ended up using the CDN for the pages that used dynamic html.

**Reply by silicahd** [reply]
I found a way.  You just need to create a command https://www.educative.io/answers/how-to-create-a-custom-artisan-command-in-laravel

      namespace App\Console\Commands;
  
     use Illuminate\Console\Command;
     use Modules\AdminPanel\Entities\Content;
     use Illuminate\Support\Facades\File;
  
     class PurgeCss extends Command
     {
      /**
       * The name and signature of the console command.
       *
       * @var string
       */
      protected $signature = "builder:purge";
  
      /**
       * The console command description.
       *
       * @var string
       */
      protected $description = 'Command description';
  
      /**
       * Create a new command instance.
       *
       * @return void
       */
      public function __construct()
      {
          parent::__construct();
      }
  
      /**
       * Execute the console command.
       *
       * @return int
       */
      public function handle()
      {
  
  
          $this->info('Importing Data from the DB');
  
          $forms = Content::where('status' , 'Active')->pluck('description', 'id');
  
          $tempDirectory = resource_path('temp');
  
          if (! File::exists($tempDirectory)) {
              File::makeDirectory(
                  $tempDirectory
              );
          }else{
              File::cleanDirectory($tempDirectory);
          }
  
          $this->info('Copy Data from the DB');
  
          $forms->each(function ($form, $id) use ($tempDirectory) {
              $file = $tempDirectory . '/form-builder-' . $id . '.blade.php';
              if (! File::exists($file)) {
                  file_put_contents(
                      $file,
                      $form
                  );
              }
          });
  
          $this->info('purging contents...');
  
          // run purge command
          shell_exec('npm run prod');
  
          $this->info('You are all set!');
          return 0;
      }
      }`

**Reply by pySilver** [reply]
@silicahd yeah, this is an easy way to accomplish the task. Ideally I'd love to avoid creating any files and use raw input instead (in multi-server setup, I rather keep things idempotent)

**Comment by silicahd** [comment]
If anybody has a way to do this I am still looking for the answer.
